qualtran/bloqs/rotations/hamming_weight_phasing_test.py

```diff
@@ -39,3 +48,58 @@ def test_hamming_weight_phasing(n: int, theta: float):
     hw_phasing = cirq.Circuit(state_prep, HammingWeightPhasing(n, theta).on(*gh.quregs['x']))
     hw_final_state = sim.simulate(hw_phasing).final_state_vector
     assert np.allclose(expected_final_state, hw_final_state, atol=1e-7)
+
+
+@attrs.frozen
```

Comment:
Wait, what's the point of this test class, why not just use the implemented bloq?

Reply:
This `HammingWeightPhasingViaPhaseGradient` class assumes that a phase gradient register is correctly prepared. The test class encapsulates the circuit to prepare a phase gradient register, apply rotations via hamming weight phasing, and then clean up the phase gradient register. 

The test class exists mainly for convenience.